### PR TITLE
Build both focal and jammy images.

### PR DESCRIPTION
**Pull request by luca-heltai**
Build both focal and jammy versions of the master branch.

### CODE REVIEW

**Comment by marcfehling** [comment]
https://github.com/dealii/dealii/blob/c0a88eee4f13a4dfb3a47d1c0241ba84193b3862/.github/workflows/docker.yml#L15

I changed this line in #15060 to make sure we build the focal image on ubuntu-focal.

When building the jammy image, I would suggest to create an entirely new job and specifically ask for `runs-on: ubuntu-22.04` (if I understood how to build docker images correctly).

**Comment by tjhei** [comment]
> I changed this line in #15060 to make sure we build the focal image on ubuntu-focal.

Why is that important? It shouldn't matter what the host system is.

**Comment by marcfehling** [comment]
> Why is that important? It shouldn't matter what the host system is.

You are right, it doesn't matter. My conception on how to build docker images was wrong, and it's clear now after reading more about their creation. It's only important what's in the image/container, and not on the host.

@luca-heltai Could you change this line back to `ubuntu-latest`? Sorry about that.

**Review by marcfehling — APPROVED** [review]

File: contrib/docker/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+ARG VER=master
+
 USER root
-RUN cd /usr/src \
-    && git clone https://github.com/dealii/dealii \
-    && cd dealii && mkdir build && cd build \
-    && cmake -GNinja \
+RUN cd /usr/src && \
+    git clone https://github.com/dealii/dealii dealii-$VER && \
+    cd dealii-$VER && \
```
Comment (marcfehling) [inline contrib/docker/Dockerfile:12]:
```suggestion
    git clone https://github.com/dealii/dealii && \
    cd dealii && \
```

We ship each image with only one version of deal.II, right? To keep all images in the same structure, I would suggest to keep the filenames of the source directories the same as well.

I like the `VER` argument though! It's an easy way to switch between branches/versions.

Comment (luca-heltai) [reply contrib/docker/Dockerfile:12]: We only ever use the source directory when debugging. The actual library is installed in /usr/local for all images, and that's all we really need to know about.  I think there is value in distinguishing one version from the other and to allow multiple versions of the source, but if you feel strongly about this, I'll make the change you suggest.

Comment (marcfehling) [reply contrib/docker/Dockerfile:12]: No, I don't feel strongly about it.

**Review by marcfehling — APPROVED** [review]

**Pull request merged**